src/support/allocators/pool.h

```diff
@@ -164,6 +164,26 @@ class PoolResource final
         m_allocated_chunks.emplace_back(m_available_memory_it);
     }
 
+    /**
+     * Return the number of actual allocation bytes, copied from memusage.h MallocUsage().
+     */
+    size_t m_system_alloc_bytes{0};
+
+    // copied from src/memusage.h
+    size_t MallocUsage(size_t alloc)
+    {
+        // Measured on libc6 2.19 on Linux.
+        if (alloc == 0) {
+            return 0;
+        } else if (sizeof(void*) == 8) {
+            return ((alloc + 31) >> 4) << 4;
+        } else if (sizeof(void*) == 4) {
+            return ((alloc + 15) >> 3) << 3;
+        } else {
+            assert(0);
+        }
+    }
```

Comment:
Copied this from https://github.com/bitcoin/bitcoin/blob/a13f3746dccd9c4ec16d6bfe9b33ebd26e3238e1/src/memusage.h#L50 because including that file would create a circular dependency. Is there a better way to do this?